### PR TITLE
Added -publish to codecoverage package generation and fixed package name

### PR DESCRIPTION
**Pull request by adityapatwardhan**
Added -Publish to Start-PSBuild used by CodeCoverage configuration so that
 .netcore dependencies are also included in the package. This makes the
 package usable on systems that do not have .netcore in path.

 Fixed the name of the coverage package by making it consistent with other
 package names.

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
@JamesWTruher @daxian-dbw  Please have a look.

**Comment by JamesWTruher** [comment]
changes look fine

**Comment by JamesWTruher** [comment]
:shipit:

File: tools/appveyor.psm1 [hunk 1]
```diff
@@ -374,6 +374,12 @@ function Compress-CoverageArtifacts
     [System.IO.Compression.ZipFile]::CreateFromDirectory($resolvedPath, $zipOpenCoverPath) 
     $null = $artifacts.Add($zipOpenCoverPath)
 
+    $name = git describe
+
+    # Remove 'v' from version, prepend 'PowerShell' - to be consistent with other package names
+    $name = $name -replace 'v',''
+    $name = 'PowerShell_' + $name
+
```
Comment (daxian-dbw) [inline tools/appveyor.psm1]: This code looks duplicate between `Compress-CoverageArtifacts` and `Invoke-AppveyorFinish`. Can we put it in a function and call that function instead?

File: test/tools/OpenCover/OpenCover.psm1 [hunk 2]
```diff
@@ -353,7 +353,8 @@ function Install-OpenCover
         throw "Download failed: $packageUrl"
     }
     
-    import-module Microsoft.PowerShell.Archive
+    ## We add ErrorAction as we do not have this module on PS v4 and below. Calling import-module will throw an error otherwise.
+    import-module Microsoft.PowerShell.Archive -ErrorAction SilentlyContinue
```
Comment (daxian-dbw) [inline test/tools/OpenCover/OpenCover.psm1]: The commit that made this change has already been merged, can you please rebase your branch?

**Comment by adityapatwardhan** [comment]
@daxian-dbw Addressed CR comments

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**